sympy/matrices/matrices.py

```diff
@@ -27,7 +27,10 @@
 
 def _iszero(x):
     """Returns True if x is zero."""
-    return x.is_zero
+    if hasattr(x, "free_symbols") and not x.free_symbols:
```

Comment:
Actually, I think, if x does not have `free_symbols`, value of `hasattr(x, "free_symbols") and not x.free_symbols` will False and we use else-branch.